src/core/environment.go

```diff
@@ -560,13 +566,54 @@ func diskPartitions() (partitions []*proto.DiskPartition) {
 	return partitions
 }
 
-func releaseInfo() (release *proto.ReleaseInfo) {
+func releaseInfo(osReleaseFile string) (release *proto.ReleaseInfo) {
+	hostReleaseInfo := getHostReleaseInfo()
+	if hostReleaseInfo == nil {
+		hostReleaseInfo = &proto.ReleaseInfo{}
+	}
```

Comment:
this re-assignment feels unusual, and it's allocating some memory that's only purpose is to prevent a nil-deref panic.

How about we change the control flow a little here, such that we only call `mergeHostAndOsReleaseInfo` if `hostReleaseInfo` is non-nil?

Reply:
Yeah your are spot-on. One of the thought process was to prevent nil-deref panic. 

In addition to that, I was thinking along the  code readability.

Let say `hostReleaseInfo == nil `, then `hostReleaseInfo` will have some memory
Thereafter, if  `err !=nil` while reading from the release file. I can simply return `hostReleaseInfo` allocated above.

Else, I call `mergeHostAndOsReleaseInfo` on `hostReleaseInfo`(may or may not be empty) and  `osrealase`